### ISSUE TITLE
tf.keras.metrics.TruePositives() returning wrong value in model.fit() when passed as metric to model.compile()

### ISSUE THREAD

**Issue by ntobis** [tensorflow/tensorflow#31513]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): MacOS
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: N/A
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below): 2.0.0-beta1
- Python version: 3.7.2
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

You can collect some of this information using our environment capture
[script](https://github.com/tensorflow/tensorflow/tree/master/tools/tf_env_collect.sh)
You can also obtain the TensorFlow version with: 1. TF 1.0: `python -c "import
tensorflow as tf; print(tf.GIT_VERSION, tf.VERSION)"` 2. TF 2.0: `python -c
"import tensorflow as tf; print(tf.version.GIT_VERSION, tf.version.VERSION)"`

**Describe the current behavior**
When I pass one-hot encoded labels as train and validation data into tensorflow keras' model.fit() function, the metric `tf.keras.metrics.TruePositives()` (and TN, FN, FP) return wrong values.
If
`train_labels` is this: `array([[1, 0], [1, 0], [0, 1]])`

and the resulting `y_pred`'s are `array([[1, 0], [1, 0], [0, 1]])`

then the sum of
```
tf.keras.metrics.TruePositives()
tf.keras.metrics.TrueNegatives() 
tf.keras.metrics.FalsePositives() 
tf.keras.metrics.FalseNegatives()
```
is 6.

**Describe the expected behavior**
The sum should be 3.

**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.
```
import tensorflow as tf
from tensorflow.keras import models, layers
import numpy as np


# Generate Data
train_data = np.random.randint(0, 100, size=(10, 215, 215, 1))
train_labels = np.eye(2)[np.random.randint(0, 2, size=(10, 1)).reshape(-1)]


# Set up metrics
metrics = ['accuracy',
           tf.keras.metrics.TruePositives(),
           tf.keras.metrics.TrueNegatives(),
           tf.keras.metrics.FalseNegatives(),
           tf.keras.metrics.FalsePositives()]

# Set up model type
model = models.Sequential(name='CNN')

# Add layers
model.add(layers.Conv2D(filters=32, kernel_size=(5, 5), strides=(2, 2), input_shape=train_data[0].shape))
model.add(layers.Conv2D(filters=64, kernel_size=(5, 5), strides=(2, 2)))
model.add(layers.Conv2D(filters=128, kernel_size=(5, 5), strides=(2, 2)))
model.add(layers.Flatten())
model.add(layers.Dense(units=128))
model.add(layers.BatchNormalization())
model.add(layers.ReLU())
model.add(layers.Dense(units=2, activation='sigmoid'))
model.compile('sgd', 'binary_crossentropy', metrics)

history = model.fit(train_data, train_labels, batch_size=1, epochs=30)
```
**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.

This also impacts recall, precision, etc. which are not correctly calculated as a result.

**Comment by ntobis** [comment]
Ok I did some more experimenting and it is fixed when the input is not 1-hot encoded and there is only 1 output neuron. So at this point I am not sure if this is intended behaviour, but all metrics run correctly if we change the following 2 lines:

This: `train_labels = np.eye(2)[np.random.randint(0, 2, size=(10, 1)).reshape(-1)]`
To: `train_labels = np.random.randint(0, 2, size=(10, 1))`
and
This: `model.add(layers.Dense(units=2, activation='sigmoid'))`
To: `model.add(layers.Dense(units=1, activation='sigmoid'))`

**Comment by jvishnuvardhan** [comment]
Automatically closing this out since I understand it to be resolved, but please let me know if I'm mistaken.Thanks!
